lib/src/data_stream/stream_writer.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:fixnum/fixnum.dart';
+
+import 'package:livekit_client/src/core/engine.dart';
+import 'package:livekit_client/src/types/other.dart';
+import '../proto/livekit_models.pb.dart' as lk_models;
+import '../types/data_stream.dart';
+
+class BaseStreamWriter<T, InfoType extends BaseStreamInfo> {
+  final StreamWriter<T> writableStream;
+  Function()? onClose;
+
+  final InfoType info;
+
+  BaseStreamWriter(
+      {required this.writableStream, required this.info, this.onClose});
+
+  Future<void> write(T chunk) async {
+    return writableStream.write(chunk);
```

Comment:
we also need an implementation for `ByteStreamWriter` where the `write` method automatically chunks oversized byte payloads to a max size kStreamChunkSize